core/lib/zksync_core/src/state_keeper/io/seal_logic.rs

```diff
@@ -186,6 +192,26 @@ impl UpdatesManager {
             extractors::display_timestamp(l1_batch_env.timestamp)
         );
 
+        let (deduplicated_writes, protective_reads): (Vec<_>, Vec<_>) = deduped_log_queries
+            .into_iter()
+            .partition(|log_query| log_query.rw_flag);
+
+        let l2_to_l1_messages =
+            extract_long_l2_to_l1_messages(&finished_batch.final_execution_state.events);
+        let initial_bootloader_contents =
+            finished_batch.final_bootloader_memory.unwrap_or_default();
+
+        let full_bootloader_memory = expand_bootloader_contents(&initial_bootloader_contents);
+
+        let bootloader_initial_content_commitment =
+            initial_heap_content_commitment_fixed(&full_bootloader_memory);
+
+        let events_queue_commitment = events_queue_commitment_fixed(
+            &finished_batch
+                .final_execution_state
+                .deduplicated_events_logs,
+        );
```

Comment:
i'd prefer to have this logic inside vm integration and don't spread the info about bootloader too much